napalm/base/test/helpers.py

```diff
@@ -4,23 +4,26 @@
 def test_model(model, data):
     """Return if the dictionary `data` complies with the `model`."""
     # Access the underlying schema for a TypedDict directly
-    model = model.__annotations__
-    same_keys = set(model.keys()) == set(data.keys())
+    annotations = model.__annotations__
+    if model.__total__:
+        same_keys = set(annotations.keys()) == set(data.keys())
+    else:
+        same_keys = set(data.keys()) <= set(annotations.keys())
```

Comment:
We generally haven't allowed there to be a subset of keys in data. Is there a way in testing that we can specifically allow this in this case.

The equivalent of:

```python
def test_model(model, data, allow_subset=False):
```

But where we pass in `allow_subset=True`, not sure how we would actualize that in the testing code.